docusaurus/docs/get-started/anatomy-of-a-plugin.md

```diff
@@ -0,0 +1,197 @@
+
+### Call resource
+
+Custom endpoints allow a data source plugin to expose custom HTTP API routes for server-side functionality. This is particularly useful when dealing with authentication, advanced queries, or processing large datasets. Custom endpoints can be created in the backend by using the `CallResourceHandler` method to handle requests and respond with data or status information.
+
+For an example of implementing custom endpoints, see the [App with Backend example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-backend/pkg/plugin/resources.go).
+
+## Panel plugins
+
+Panel plugins extend Grafana by providing custom widgets to visualize data or provide other useful functionality within dashboards.
+
+![The components of a Panel plugin](./images/panel-plugin.png)
+
+### Visualization
+
+Panel plugins provide visual representations of data in Grafana dashboards. To create a custom visualization, developers use React components to define how data will be rendered on the dashboard. This visualization can be anything from a simple chart to a complex interactive widget. The panel’s `render()` function defines how the data is passed into the visualization and how updates are handled when data or options change.
+
+For more details on panel visualizations, you can refer to the [panel plugin example](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/panel-basic).
+
+### Panel Options
+
+Panel options allow users to customize the behavior and appearance of the panel plugin. Developers can define these options by implementing the `OptionsEditor` component, where they can expose options relevant to the visualization. These options are passed into the panel’s `render()` function, allowing for dynamic updates based on user inputs.
+
+You can see an example of how to implement panel options in the [basic panel example](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/panel-basic/src/types.ts#L5).
+
+## Plugin folder structure
+
+After running the `create-plugin` tool, a new folder for your plugin will be generated. The plugin folder follows a standard naming convention (e.g. `organization-pluginName-pluginType`) and contains all the necessary files for building, running, and testing your plugin. Here's an overview of the folder layout and key files:
+
+```
+myorg-myplugin-datasource/
+├── .config/
+├── .eslintrc
+├── .github
+│   └── workflows
+├── .gitignore
+├── .nvmrc
+├── .prettierrc.js
+├── CHANGELOG.md
+├── LICENSE
+├── Magefile.go
+├── README.md
+├── docker-compose.yaml
+├── go.mod
+├── go.sum
+├── jest-setup.js
+├── jest.config.js
+├── node_modules
+├── package.json
+├── pkg
+│   ├── main.go
+│   └── plugin
+├── playwright.config.ts
+├── src
+│   ├── README.md
+│   ├── components
+│   ├── datasource.ts
+│   ├── img
+│   ├── module.ts
+│   ├── plugin.json
+│   └── types.ts
+├── tsconfig.json
+└── tests
```

Comment:
Nit: This structure is constantly updated/changed so to prevent the risk of having this documentation out of sync an alternative would be to just highlight the most important folders. 

Alternative add a note to this structure saying something like "Minor differences between this documentation and what will be scaffolded by create-plugin is expected."

Reply:
Yeah I agree @mckn, I went back and forth on whether to include this verbatim or not. I'll have a think and push an update today. We highlight the key files below this and mention the main folders for devs to be aware of (`src`, `pkg`, and `tests`), so it may be that we can remove this entirely.